bindgen/clang.rs

```diff
@@ -1183,20 +1193,29 @@ pub(crate) enum LayoutError {
     /// Asked for the layout of a field in a type that does not have such a
     /// field.
     InvalidFieldName,
+    /// The type is undeduced.
+    Undeduced,
     /// An unknown layout error.
     Unknown,
 }
 
-impl ::std::convert::From<i32> for LayoutError {
+impl From<i32> for LayoutError {
     fn from(val: i32) -> Self {
+        Self::from(i64::from(val))
+    }
+}
+
+impl From<i64> for LayoutError {
+    fn from(val: i64) -> Self {
         use self::LayoutError::*;
 
-        match val {
-            CXTypeLayoutError_Invalid => Invalid,
-            CXTypeLayoutError_Incomplete => Incomplete,
-            CXTypeLayoutError_Dependent => Dependent,
-            CXTypeLayoutError_NotConstantSize => NotConstantSize,
-            CXTypeLayoutError_InvalidFieldName => InvalidFieldName,
+        match val.try_into() {
+            Ok(CXTypeLayoutError_Invalid) => Invalid,
+            Ok(CXTypeLayoutError_Incomplete) => Incomplete,
+            Ok(CXTypeLayoutError_Dependent) => Dependent,
+            Ok(CXTypeLayoutError_NotConstantSize) => NotConstantSize,
+            Ok(CXTypeLayoutError_InvalidFieldName) => InvalidFieldName,
+            Ok(CXTypeLayoutError_Undeduced) => Undeduced,
```

Comment:
This adds a new option, unsure whether that was intentional

Reply:
Yeah, this PR was cherry-picked from a commit in another branch. This addition should probably go in a separate PR.

Reply:
Ah, I needed to add it in order to fully implement `fallible_size`/`fallible_align`, which can return this error. Previously, only the error number `-6` was returned in `clang_size_of`/`clang_align_of` instead of a proper error type.